### ISSUE TITLE
TagViewer: "Unable to fetch required data" on any post

### ISSUE THREAD

**Issue by andrew-jc-hsu** [new-xkit/XKit#1719]
"TagViewer could not get the required data from the Tumblr servers.

Please try again later or file a bug at the New Xkit blog."
error displays when trying to use TagViewer to view any post tested.

**Comment by hobinjk** [comment]
Able to reproduce on my janky test account with:
```bash
curl 'https://www.tumblr.com/svc/tumblelog/only1600kids/187558336449/notes?mode=reblogs_with_tags' -H 'sec-fetch-mode: cors' -H 'accept-encoding: gzip, deflate, br' -H 'accept-language: en-US,en;q=0.9' -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36' -H 'accept: */*' -H 'referer: https://www.tumblr.com/dashboard' -H 'authority: www.tumblr.com' -H 'sec-fetch-site: cross-site' --compressed
```

**Comment by hobinjk** [comment]
Narrowed it down to just the "mode=reblog_with_tags", omitting that makes it return a non-500

**Comment by AprilSylph** [comment]
and thats the part we rely on, fuck. but it's also what tag crawler relies on, is tag crawler broken where tagviewer is too?

**Comment by AprilSylph** [comment]
...wait, could it be that this endpoint is only available to tag crawler users?

**Comment by hobinjk** [comment]
Only TagViewer uses the `?mode=reblogs_with_tags`, activity+ uses `?mode=all`. I have no idea why one would work while the other doesn't

**Comment by hobinjk** [comment]
i.e. https://www.tumblr.com/svc/tumblelog/only1600kids/187558336449/notes?mode=reblogs_with_tags doesn't work

https://www.tumblr.com/svc/tumblelog/only1600kids/187558336449/notes?mode=all does work

**Comment by hobinjk** [comment]
(doesn't work in curl/fetch, does work in-browser for that extra spice)

**Comment by hobinjk** [comment]
The request needs some cookies to succeed that are valid on my main account but invalid on my janky testing account. Most notable cookies missing from the happy request are `activity_range=threedays` and `pfg=<long string>`

**Comment by hobinjk** [comment]
@AprilSylph You were right! The 500 went away when I enabled the tag crawler tumblr labs feature

**Comment by AprilSylph** [comment]
god dammit

**Comment by AprilSylph** [comment]
i guess the proper solution to this would be to switch back to trawling all notes for reblogs but then making a peepr request per reblog instead of an API request, so we can still show tags from dash-only blogs and respect blocks, since having a Tumblr Labs feature enabled is admittedly a weird requirement for an XKit extension (as much as I'd love to just tell everyone to use tag crawler instead of tagviewer)

hopefully i'll get around to this tomorrow

**Comment by AprilSylph** [comment]
resolved by #1720
